plugins/@grouparoo/google-sheets/src/lib/sheet-import/downloadAndRefreshFile.ts

```diff
@@ -0,0 +1,63 @@
+import { log } from "actionhero";
+import fs from "fs-extra";
+import path from "path";
+import { tmpdir } from "os";
+import { SimpleSourceOptions } from "@grouparoo/core";
+import Spreadsheet from "./Spreadsheet";
+import csvWriter from "csv-write-stream";
+
+const CSV_CACHE_MILLISECONDS = 1000 * 60;
+
+export async function downloadAndRefreshFile(
+  sourceId: string,
+  appOptions: SimpleSourceOptions,
+  sourceOptions: SimpleSourceOptions
+) {
+  const sheet = new Spreadsheet(
+    appOptions,
+    sourceOptions.sheet_url?.toString()
+  );
+
+  const localDir = path.join(tmpdir(), "google-sheets-cache", sourceId);
+  const localPath = path.join(localDir, `${sheet.docId}.csv`);
```

Comment:
👍 using sourceId makes these unique if you have more than one google sheet source